conanfile.py

```diff
@@ -189,62 +190,67 @@ def patch_misc_files(self):
 
         if self.settings.os == "Macos" and self.options.darwin_ssl:
-            suffix +=  "--with-darwinssl "
+            params.append("--with-darwinssl")
         elif self.settings.os == "Windows" and self.options.with_winssl:
-            suffix += " --with-winssl "
+            params.append("--with-winssl")
         elif self.options.with_openssl:
             openssl_path = self.deps_cpp_info["OpenSSL"].rootpath.replace('\\', '/')
-            suffix += " --with-ssl=%s " % openssl_path
+            params.append("--with-ssl=%s" % openssl_path)
         else:
-            suffix += " --without-ssl "
+            params.append("--without-ssl")
 
         if self.options.with_libssh2:
-            suffix += "--with-libssh2=%s " % self.deps_cpp_info["libssh2"].lib_paths[0].replace('\\', '/')
+            params.append("--with-libssh2=%s " % self.deps_cpp_info["libssh2"].lib_paths[0].replace('\\', '/'))
         else:
-            suffix += " --without-libssh2 "
+            params.append("--without-libssh2")
 
-        suffix += "--with-zlib=%s " % self.deps_cpp_info["zlib"].lib_paths[0].replace('\\', '/')
+        params.append("--with-zlib=%s " % self.deps_cpp_info["zlib"].lib_paths[0].replace('\\', '/'))
 
         if not self.options.shared:
-            suffix += " --disable-shared"
-            suffix += " --enable-static"
+            params.append("--disable-shared")
+            params.append("--enable-static")
         else:
-            suffix += " --enable-shared"
-            suffix += " --disable-static"
+            params.append("--enable-shared")
+            params.append("--disable-static")
 
         if self.options.disable_threads:
-            suffix += " --disable-thread"
+            params.append("--disable-thread")
 
         if not self.options.with_ldap:
-            suffix += " --disable-ldap"
+            params.append("--disable-ldap")
 
         if self.options.custom_cacert:
-            suffix += ' --with-ca-bundle=cacert.pem'
+            params.append('--with-ca-bundle=cacert.pem')
 
-        suffix += ' --prefix=%s' % self.package_folder.replace('\\', '/')
+        params.append('--prefix=%s' % self.package_folder.replace('\\', '/'))
 
         # for mingw
         if self.is_mingw:
             if self.settings.arch == "x86_64":
-                suffix += ' --build=x86_64-w64-mingw32'
-                suffix += ' --host=x86_64-w64-mingw32'
+                params.append('--build=x86_64-w64-mingw32')
+                params.append('--host=x86_64-w64-mingw32')
             if self.settings.arch == "x86":
-                suffix += ' --build=i686-w64-mingw32'
-                suffix += ' --host=i686-w64-mingw32'
+                params.append('--build=i686-w64-mingw32')
+                params.append('--host=i686-w64-mingw32')
+
+        # Cross building flags
+        if tools.cross_building(self.settings):
+            if self.settings.os == "Linux" and "arm" in self.settings.arch:
```

Comment:
this is not a sane assumption
see https://wiki.debian.org/ArmPorts

Reply:
This is a really good point ...

So, considering Conan supports:
`arch: [x86, x86_64, ppc64le, ppc64, armv6, armv7, armv7hf, armv8, sparc, sparcv9, mips, mips64, avr, armv7s, armv7k]`

```
def get_linux_arm_host(self):
arch = 'arm-linux-gnu'
# aarch64 could be added by user
if 'aarch64' in self.settings.arch:
    arch = 'aarch64-linux-gnu'
elif 'arm' in self.settings.arch and 'hf' in self.settings.arch:
    arch = 'arm-linux-gnueabihf'
elif 'arm' in 'aarch64-linux-gnu'
:
    
```

Reply:
yep, something like that. not sure how to handle regular arm vs eabi though.

Reply:
> ArmEabiPort - newer port using the "new" ABI (EABI), supported on Arm v4t and higher. First released with 5.0 (lenny). GNU Triplet: arm-linux-gnueabi 

maybe we could consider >armv4 is eabi supported

Reply:
Any idea @lasote?

Reply:
Good point. As I said in one of my original PRs, I realize that the AutotoolsBuildEnvironment has to be improved in some ways and opened an issue. I would like to use the current code of triplets that the helper already has. The current implementation is a private function doing this, that, by the way, probably it is not perfect, so feel free to contribute ;)

```python

 def _get_triplet(self, the_arch, the_os):
        """
        machine-vendor-op_system, But vendor can be omitted in practice
        """

        # Calculate the arch
        machine = {"x86": "i686" if the_os != "Linux" else "x86",
                   "x86_64": "x86_64",
                   "armv6": "arm",
                   "armv7": "arm",
                   "armv7s": "arm",
                   "armv7k": "arm",
                   "armv7hf": "arm",
                   "armv8": "aarch64"}.get(the_arch, None)
        if machine is None:
            self._conanfile.output.warn("Unknown '%s' machine, Conan doesn't know how to "
                                        "translate it to the GNU triplet, please report at "
                                        " https://github.com/conan-io/conan/issues" % the_arch)
            return "unknown"

        # Calculate the OS
        compiler = self._conanfile.settings.get_safe("compiler")
        if compiler == "gcc":
            windows_op = "w64-mingw32"
        elif compiler == "Visual Studio":
            windows_op = "windows-msvc"
        else:
            windows_op = "windows"

        op_system = {"Windows": windows_op,
                     "Linux": "linux-gnu",
                     "Darwin": "apple-darwin",
                     "Android": "linux-android",
                     "Macos": "apple-darwin",
                     "iOS": "apple-darwin",
                     "watchOS": "apple-darwin",
                     "tvOS": "apple-darwin"}.get(the_os, the_os.lower())

        if the_os in ("Linux", "Android"):
            if "arm" in the_arch and the_arch != "armv8":
                op_system += "eabi"

            if the_arch == "armv7hf" and the_os == "Linux":
                op_system += "hf"

        return "%s-%s" % (machine, op_system)

```

So, This is something I really want to solve for the next conan version, this recipe will be simplified a lot and it will be much more solid. In the meantime I let you choose the approach. I'm ok if you keep the recipe compatible with a couple of arm devices or using the function you suggested.

By the way, arm8 == aarch64 afaik. let me know if it's not true.

Reply:
arm8 == aarch64 afaik - that's right

Reply:
ARM ABIs are hard, Debian is the good source of truth about them.

Due to complexity, version range and build time of the libcurl recipe it'll be much better if ARM support could be used as the helper from Conan core when #2691 is done. Testing even small changes is very slow for libcurl :(

Reply:
I read [autotools_environment](https://github.com/conan-io/conan/blob/master/conans/client/build/autotools_environment.py), so `_get_triplet` is used in our case, because we run auto tools on Linux.  
So, I'll use my suggestion to distinguish ARM platform.